clients/apple/Shared/Views/Book/FileListView.swift

```diff
@@ -122,58 +62,65 @@ struct FileListView: View {
             core.handleError(err)
         }
     }
-
-    func handleSelectFolder(meta: FileMetadata) {
-        withAnimation {
-            path.append(currentFolder)
-            currentFolder = meta
-        }
-    }
-
-    func handleNavigateUp() {
-        withAnimation {
-            path.popLast().map {
-                currentFolder = $0
-            }
+    
+    func renderCell(meta: FileMetadata) -> AnyView {
+        if meta.fileType == .Folder {
+            return AnyView (
+                NavigationLink(destination: FileListView(core: core, currentFolder: meta, account: account)) {
+                    FileCell(meta: meta)
+                    
+                }.isDetailLink(false)
+                .contextMenu(menuItems: {
+                    Button(action: {
+                        handleDelete(meta: meta)
+                    }) {
+                        Label("Delete", systemImage: "trash.fill")
+                    }
+                })
```

Comment:
Put context menu on the returned `AnyView` not each item unless we expect to have diverging menus

Reply:
fd6ff2c9